### PR TITLE
Moved go and java model generation into gen package

### PR DESCRIPTION
**Pull request by boynton**
From the ardielle-tools repo, to encourage reuse. Bumped version to
1.5.1.

### CODE REVIEW

**Comment by boynton** [comment]
The regression was introduced by b919d345f28a6a119110ec3a245c24ec25c100c5, where a generated file was hand-edited.

File: gen/genutil.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
+package gen
+
+import(
+	"bufio"
+	"bytes"
+	"os"
+   "path/filepath"
```
Comment (dmitris) [inline gen/genutil.go]: any reason this is not gofmt'ed, or why we would not want to format it?

**Review by dmitris — COMMENTED** [review]
In master version, `find . -name '*.go' | xargs -n1 gofmt -l` shows no output, but in the PR's one, I get:
```
$ find . -name '*.go' | xargs -n1 gofmt -l
./gen/gomodel/gomodel.go
./gen/gomodel/gomodel_test.go
./gen/javamodel/javamodel.go
./gen/javamodel/javamodel_test.go
./gen/javamodel/javaschema.go
./gen/genutil.go
```

Also, for the generated files, could we add on top of the files a notice/warning like `// Code generated by "[command-invocation params]"; DO NOT EDIT.` - to avoid confusion?

Would it make sense to provide a `go generate` command to run the bootstrap (and also serve as documentation)?

File: gen/genutil.go (hunk 1 above)
Comment (boynton) [reply gen/genutil.go]: No reason, I overlooked it. I'll fix it

**Comment by boynton** [comment]
The go formatting (and go vet) were not run on those new files. This is fixed now.

The lack of warning was only on the generated xxx_schema.go file generation, that is also fixed.

The whole bootstrap could be made clearer, but I think it is more than just go generate. Part of this PR is to move model generation into this library package, from the tooling package. I find that keeping them in sync for bootstrapping is a pain when in separate repos. But that fix would be for another time. All of the codegen stuff needs a cleanup, if anybody has time to contribute :-)

**Review by dmitris — APPROVED** [review]
👍

**Pull request merged**